src/pyhf/parameters/paramsets.py

```diff
@@ -2,25 +2,34 @@
 
 
 class paramset(object):
+    __slots__ = ['n_parameters', 'suggested_init', 'suggested_bounds', 'fixed']
```

Comment:
why do we need `__slots__`?

Reply:
Only reason really is to save space and ensure we're not adding any other extra attributes to the paramsets. This also helps automatically document all the available properties on a paramset. In this case, it simply formalizes what we're already doing.